packages/astro/src/core/errors/errors-data.ts

```diff
@@ -475,6 +477,30 @@ See https://docs.astro.build/en/guides/server-side-rendering/ for more informati
 		title: 'Failed to parse Markdown frontmatter.',
 		code: 6001,
 	},
+	/**
+	 * @docs
+	 * @message
+	 * **Example error message:**<br/>
+	 * Could not parse frontmatter in **blog** → **post.md**<br/>
+	 * "title" is required.<br/>
+	 * "date" must be a valid date.
+	 * @description
+	 * A Markdown document's frontmatter in `src/content/` does not match your collection schema.
+	 * Make sure that all required fields are present, and that all fields are of the correct type.
+	 * You can check against the collection schema in your `src/content/config` file.
+	 * See the [Content collections documentation](https://docs.astro.build/en/guides/content-collections) for more information.
+	 */
+	MarkdownContentSchemaValidationError: {
+		title: 'Content collection frontmatter invalid.',
+		code: 6002,
+		message: (collection: string, entryId: string, error: ZodError) => {
+			return [
+				`${String(collection)} → ${String(entryId)} frontmatter does not match collection schema.`,
+				...error.errors.map((zodError) => zodError.message),
+			].join('\n');
+		},
+		hint: 'See https://docs.astro.build/en/guides/content-collections for more information on content schemas.',
+	},
```

Comment:
Flagging docs team for review!

Reply:
This looks like a very helpful error message @bholmesdev!

nit (non-blocking): "...does not match ITS collection schema"  ? 
(Since the doc has *a* single collection; *you* might have several collections, and therefore several collection schemas.)

question: Is `src/content/config` (with no extension) how we reference a *file*? I know that it's tricky because it could have one of a few different file extensions. At a glance, that *looks* like a folder to me, not a file, without any file extension so I did want to point it out. I'll leave it to your judgement whether to use `.ts` here, with the understanding that it's a placeholder for whichever file extension is use (as we sort of do in the docs page) or whether you think it's fine!

Reply:
Totally fair! Since we document `.ts` as the convention, we can stick with that here. The galaxy brain approach would be adapting the error message to your project but... let's keep it simple.

And yes, will change to `it's`